motor.c

```diff
@@ -305,3 +319,30 @@ void watchdog_init(void) {
 	IWDG_SetReload(2); // 187.5us; for some reason, a value of 1 don't work, only 2
 	IWDG_ReloadCounter();
 }
+
+
+// Move these includes here if you want to see the identifiers that the ISR code uses
+// without much regard for safety (most of whats left (except those timers) are effectively constants (except if adjusted during debug).
+//#include "ACAcontrollerState.h"
+//#include "ACAcommons.h"
+
+
+// The concept here is loaned from linux; except that SDCC doesnt do typeof(x), so we need one per width.
+#define READ_ONCE_U16(x)	(*(const volatile uint16_t *)&(x))
+#define WRITE_ONCE_U16(x, val)						\
+do {									\
+	*(volatile uint16_t *)&(x) = (val);				\
+} while (0)
+
+/* Communicate between "fast loop" (ISR) and rest of the system. */
+void motor_slow_update_pre(void) {
+	disableInterrupts();
+	ui16_motor_speed_erps = READ_ONCE_U16(motor_speed_erps);
```

Comment:
reads internally set __motor_speed_erps__ and sets it for extern __ui16_motor_speed_erps__